Flappy Bird Clone/Assets/Scripts/Obstacles/ObstacleBehaviour.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using Player;
+using UnityEngine;
+
+namespace Obstacles
+{
+    public class ObstacleBehaviour : MonoBehaviour
+    {
+        // DELEGATES
+        public delegate void PlayerTouchedObstacle();
+
+        // EVENTS
+        public static event PlayerTouchedObstacle OnPlayerTouchedObstacle;
+        
+        // VARIABLES
+        private BoxCollider2D _collider;
+        
+        // METHODS
+        private void Awake()
+        {
+            _collider = GetComponent<BoxCollider2D>();
+        }
+
+        // EVENTS
+        private void OnCollisionEnter2D(Collision2D other)
+        {
+            if (_collider.isTrigger || !other.gameObject.TryGetComponent(out PlayerBehaviour player))
```

Comment:
🤔 if it's trigger it collides and enter in this function? I don't remember.. heh

Reply:
It only activates if the object isn't a trigger, added this check in case of colliding with a 'IsTrigger' collider (for any reason)